wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -256,21 +263,46 @@ instance ByteArray.ByteArrayAccess a => ByteArray.ByteArrayAccess (V1 a) where
    length (V1 a) = ByteArray.length a
    withByteArray (V1 a) callback = ByteArray.withByteArray a callback
 
-instance Arbitrary (V1 (Mnemonic 12)) where
+instance
+    ( n ~ EntropySize mw
+    , mw ~ MnemonicWords n
+    , ValidChecksumSize n csz
+    , ValidEntropySize n
+    , ValidMnemonicSentence mw
+    , Arbitrary (Entropy n)
+    ) => Arbitrary (V1 (Mnemonic mw)) where
     arbitrary =
         V1 <$> arbitrary
 
-instance ToJSON (V1 (Mnemonic 12)) where
+instance
+    ( n ~ EntropySize mw
+    , mw ~ MnemonicWords n
+    , ValidChecksumSize n csz
+    , ValidEntropySize n
+    , ValidMnemonicSentence mw
+    ) => ToJSON (V1 (Mnemonic mw)) where
     toJSON =
         toJSON . unV1
 
-instance FromJSON (V1 (Mnemonic 12)) where
+instance
+    ( n ~ EntropySize mw
+    , mw ~ MnemonicWords n
+    , ValidChecksumSize n csz
+    , ValidEntropySize n
+    , ValidMnemonicSentence mw
+    ) => FromJSON (V1 (Mnemonic mw)) where
     parseJSON =
         fmap V1 . parseJSON
 
-instance ToSchema (V1 (Mnemonic 12)) where
+instance
+    ( mw ~ MnemonicWords n
+    , n ~ EntropySize mw
+    , ValidChecksumSize n csz
+    , ValidEntropySize n
+    , ValidMnemonicSentence mw
+    ) => ToSchema (V1 (Mnemonic mw)) where
     declareNamedSchema _ = do
-        NamedSchema _ schm <- declareNamedSchema (Proxy @(Mnemonic 12))
+        NamedSchema _ schm <- declareNamedSchema (Proxy @(Mnemonic mw))
         return $ NamedSchema (Just "V1BackupPhrase") schm
```

Comment:
What about getting back to plain types and also use a newtype for the backup phrase after all? Having instances on the newtype instead of treating all mnemonics the same way. 
Can do in a different PR though.

Reply:
easy enough to do now :)